src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,109 @@
+import { save, load } from '../lib/save-load.js';
+
+export default class ActivityLog {
+  constructor() {
+    this.loadLogBtn = document.getElementById('loadLogBtn');
+    this.logInputFile = document.getElementById('logInputFile');
+    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.showActivityLog = document.getElementById('showActivityLog');
+    this.showLoadedLog = document.getElementById('showLoadedLog');
+    this.tableFirstRow = document.querySelector('tbody tr:first-child');
+  }
+
+  logs = [];
+  loadedLogs = [];
+
+  async getExistingLogs() {
+    const { existingLogs } = await browser.runtime.sendMessage({
+      requestType: 'sendAllLogs',
+      requestTo: 'ext-monitor',
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    const loadedLogsAsText = await load.loadLogAsText(file).catch((error) => {
+      alert(error);
+    });
+    const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+    this.loadedLogs.push(...loadedJSONLogs);
+    this.showLoadedLog.textContent = loadedLogsAsText;
+  }
+
+  async saveLogs(logs) {
+    let isDownloadDone = await save.saveAsJSON(logs).catch((error) => {
+      alert(error);
+    });
+
+    if (isDownloadDone === 'complete') {
+      console.log('download complete from activitylogs page');
+    }
+  }
+
+  messageHandlers = {
+    appendLogs: ({ log }) => {
+      this.logs.push(log);
+      this.renderTable([log]);
+    },
+  };
+
+  renderTable(logs) {
+    for (const log of logs) {
+      const newRow = document.createElement('tr');
+      newRow.insertAdjacentHTML('beforeend', `<td>${log.id}</td>`);
+      newRow.insertAdjacentHTML('beforeend', `<td>${log.timeStamp}</td>`);
+      newRow.insertAdjacentHTML('beforeend', `<td>${log.type}</td>`);
+      newRow.insertAdjacentHTML('beforeend', `<td>${log.name}</td>`);
+      newRow.insertAdjacentHTML('beforeend', `<td>${log.viewType}</td>`);
```

Comment:
@atiqueahmedziad As we discussed yesterday during our weekly meeting, DOM API methods that accept non sanitized HTML content should be avoided, because they would open the monitor extension to XSS attacks originated by one of the monitored extensions.

Let's remove these ones (e.g. by creating the `td` elements and then filling them with the expected text content using the textContent property).

As I did mention in the meeting we may also add [eslint-plugin-no-unsanitized](https://github.com/mozilla/eslint-plugin-no-unsanitized) to the dev dependencies and then enable it in the eslint configuration used in this repository, as an additional way to prevent this kind of dangerous code pattern to slip into a pull request without being noticed.

Reply:
Thanks for mentioning again. I have opened pr #8 to add "eslint-plugin-no-unsanitized". :-)